gadget/device_linux.go

```diff
@@ -95,3 +100,42 @@ func encodeLabel(in string) string {
 	}
 	return buf.String()
 }
+
+// FindMountPointForStructure locates a mount point of a device that matches
+// given structure. The structure must have a filesystem defined, otherwise an
+// error is raised.
+func FindMountPointForStructure(ps *PositionedStructure) (string, error) {
+	if ps.IsBare() {
+		return "", ErrNoFilesystemDefined
+	}
+
+	devpath, err := FindDeviceForStructure(ps)
+	if err != nil {
+		return "", err
+	}
+
+	var mountPoint string
+	mountInfo, err := osutil.LoadMountInfo(filepath.Join(dirs.GlobalRootDir, osutil.ProcSelfMountInfo))
+	if err != nil {
+		return "", fmt.Errorf("cannot read mount info: %v", err)
+	}
+	for _, entry := range mountInfo {
+		if entry.MountSource != devpath || entry.FsType != ps.Filesystem {
```

Comment:
I would add a small snippet like this here:
```
if mountPoint != "" && (mountPoint == entry.MountDir || strings.HasPrefix(mountPoint, entry.MountDir + "/")) {
  // mount entry clobbers visibility to previously discovered match
  mountPoint = "" // clobbered
}
```

Reply:
There's a `break` couple of lines down, so the loop will finish once a match is found. This makes me think, I should restructure the loop bit to make the lookup easier to read.

Reply:
Pushed an update, should be clearer now.